actor/src/main/scala/org/apache/pekko/actor/javadsl/ActorSystem.scala

```diff
@@ -0,0 +1,139 @@
+import scala.concurrent.ExecutionContext
+
+import com.typesafe.config.{ Config, ConfigFactory }
+
+import org.apache.pekko
+import pekko.actor.ActorSystem.findClassLoader
+import pekko.actor._
+import pekko.actor.setup.ActorSystemSetup
+import pekko.util.FutureConverters.FutureOps
+
+trait ActorSystem extends org.apache.pekko.actor.ActorSystem {
+
+  /**
+   * Asynchronously terminates this actor system by running [[CoordinatedShutdown]] with reason
+   * [[CoordinatedShutdown.ActorSystemTerminateReason]].
+   *
+   * If `pekko.coordinated-shutdown.run-by-actor-system-terminate` is configured to `off`
+   * it will not run `CoordinatedShutdown`, but the `ActorSystem` and its actors
+   * will still be terminated.
+   *
+   * This will stop the guardian actor, which in turn
+   * will recursively stop all its child actors, and finally the system guardian
+   * (below which the logging actors reside) and then execute all registered
+   * termination handlers (see [[ActorSystem.registerOnTermination]]).
+   * Be careful to not schedule any operations on completion of the returned future
+   * using the dispatcher of this actor system as it will have been shut down before the
+   * future completes.
+   */
+  def terminateAsync(): CompletionStage[Terminated] = terminateImpl().asJava
+
+  /**
+   * Returns a [[CompletionStage]] which will be completed after the [[ActorSystem]] has been terminated
+   * and termination hooks have been executed. If you registered any callback with
+   * [[ActorSystem.registerOnTermination]], the returned Future from this method will not complete
+   * until all the registered callbacks are finished. Be careful to not schedule any operations,
+   * such as `onComplete`, on the dispatchers (`ExecutionContext`) of this actor system as they
+   * will have been shut down before this future completes.
+   */
+  override def getWhenTerminated: CompletionStage[Terminated] = whenTerminatedImpl.asJava
+}
+
+object ActorSystem {
+
+  /**
+   * Creates a new ActorSystem with the name "default",
+   * obtains the current ClassLoader by first inspecting the current threads' getContextClassLoader,
+   * then tries to walk the stack to find the callers class loader, then falls back to the ClassLoader
+   * associated with the ActorSystem class.
+   * Then it loads the default reference configuration using the ClassLoader.
+   */
+  def create(): ActorSystem = create("default")
+
+  /**
+   * Creates a new ActorSystem with the specified name,
+   * obtains the current ClassLoader by first inspecting the current threads' getContextClassLoader,
+   * then tries to walk the stack to find the callers class loader, then falls back to the ClassLoader
+   * associated with the ActorSystem class.
+   * Then it loads the default reference configuration using the ClassLoader.
+   */
+  def create(name: String): ActorSystem = create(name, None, None, None)
+
+  /**
+   * Scala API: Creates a new actor system with the specified name and settings
```

Comment:
This can't be a Scala API if it's in a Java DSL class - this appears in other places in this class too

Reply:
This was accidentally copied over